lib/class-google.php

```diff
@@ -127,50 +141,122 @@ protected function api_key_check( $key ) {
 	 */
 	protected function load_fonts() {
 		if ( empty( $this->fonts ) ) {
-			$key = $this->api_key;
 
-			if ( '' === $key ) {
+			// Try to load the cached fonts.
+			$this->fonts = $this->get_cached_fonts();
+
+			// If no cached fonts or transient is expired, load from API.
+			if ( empty( $this->fonts ) ) {
+				$this->fonts = $this->fetch_api_fonts();
+			}
+
+			// If still no fonts, try to load fonts saved in options.
+			if ( empty( $this->fonts ) ) {
+				$this->fonts = $this->get_fonts_option();
+			}
+
+			// If still no fonts, there's a problem.
+			if ( empty( $this->fonts ) ) {
 				return false;
 			}
 
-			$response = wp_remote_get( esc_url( 'https://www.googleapis.com/webfonts/v1/webfonts?key=' . $key ) );
-
-			if ( is_array( $response ) ) {
-				if ( isset( $response['response']['code'] ) && 200 === $response['response']['code'] ) {
-					$response_fonts = json_decode( $response['body'] )->items;
-					$fonts          = array();
-
-					foreach ( $response_fonts as $font ) {
-						$font_key = sanitize_title( $font->family );
-
-						$type     = $this->get_font_type( $font->category, false );
-						$alt_font = $this->get_font_type( $font->category, true );
-
-						$variants = array_map(
-							array( $this, 'map_variants' ),
-							$font->variants
-						);
-
-						$val = str_replace( ' ', '+', $font->family ) . ':' . implode( ',', $variants );
-
-						$fonts[ $font_key ] = dpp_font(
-							array(
-								'src'    => 'web',
-								'url'    => esc_url( 'https://fonts.google.com/specimen/' . str_replace( ' ', '+', $font->family ) ),
-								'label'  => $font->family,
-								'css'    => '"' . $font->family . '", ' . $alt_font,
-								'type'   => $type,
-								'source' => $this->name,
-								'val'    => $val,
-								'link'   => '//fonts.googleapis.com/css?family=' . $val,
-							)
-						);
-					}
-
-					$this->fonts = $fonts;
+			// Cache the fonts.
+			$this->cache_fonts( $this->fonts );
```

Comment:
this should be inside the fetch_api_fonts so the transit and option are updated when the API is called, not every time. Otherwise the transient won't clear.